libs-scala/ports/src/main/scala/com/digitalasset/ports/FreePort.scala

```diff
@@ -3,17 +3,127 @@
 
 package com.daml.ports
 
+import java.io.IOException
 import java.net.{InetAddress, ServerSocket}
 
+import com.daml.bazeltools.BazelRunfiles.rlocation
+
+import scala.io.Source
+import scala.sys.process.Process
+import scala.util.{Failure, Random, Try}
+
 object FreePort {
+  private val maxAttempts = 100
+  private[ports] val dynamicRange = dynamicPortRange()
 
+  /** Find a free port.
+    *
+    * This determines whether a port is free by attempting to bind it and immediately releasing it again.
+    * Note, this does not guarantee that the port is still free after this function returns, see [[LockedFreePort]].
```

Comment:
I think there is a potential race condition with the `LockedFreePort` approach, as described in the commit message [here](https://github.com/digital-asset/daml/pull/10774/commits/7805c40ae7e56f3b9dbdf2574bb3b0c7210e8846). However, that issue was already present before this PR, so I'd prefer to address it separately.